File: backend/models/course.py
# backend/models/course.py
from database.db import Base
from sqlalchemy import Column, Integer, String

class Course(Base):
    __tablename__ = 'courses'
    id = Column(Integer, primary_key=True)
    title = Column(String(200), nullable=False)
    description = Column(String(500))
    level = Column(String(50))
    xp_reward = Column(Integer, default=0)
    image_url = Column(String(500))

File: frontend/CyberLearnApp/app/src/main/java/com/example/cyberlearnapp/network/models/Course.kt
package com.example.cyberlearnapp.network.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Course(
    val id: Int,
    val title: String,
    val description: String,
    val level: String,
    @SerialName("xp_reward") val xpReward: Int,
    @SerialName("image_url") val imageUrl: String? = null
)